### ISSUE TITLE
Template Engine Agnosticism

### ISSUE THREAD

**Issue by madeofpeople** [kwizzn/metalsmith-kss#2]
Adding support for consolidate.js would allow us to build our styleguides using the same template engine as the partials we're defining in our projects.

Meaning instead of duplicating partial markup in our scss files, we just include our templates.

One less step to making sure our  KSS styleguides are up to date.
Mega maintainable!

Handlebar's support of includes being less than magical out the box.

A swig example (which would be very similar in nunjucks, and jade):
```
/*
Markup:
{% include "../templates/partials/cta.html" %}
*/

.cta-wrapper {
… our css 
}
```

**Comment by kwizzn** [comment]
You're right and as you can see from the project's README.md, metalsmith-kss should have supported consolidate.js from the start but I did not find the time, yet. However, we would gladly accept a PR if you wish to contribute!

**Comment by RobLoach** [comment]
I could be wrong, but I believe [kss-node](https://github.com/kss-node/kss-node), in its current state, targets Handlebars specifically. Would be cool to abstract that all and off-load that to Metalsmith.

**Comment by kwizzn** [comment]
Do you think this is an option for kss-node, @RobLoach?

**Comment by madeofpeople** [comment]
@kwizzn you are correct, I hadn't gotten to the very bottom of the readme.

@RobLoach interesting possibility… 
Still all implementations allow you to specify a template, but you're right that there are assumptions there that could trip us up. Maybe best to workaround for now.
